src/compiler/moduleNameResolver.ts

```diff
@@ -898,6 +903,11 @@ namespace ts {
      * in cases when we know upfront that all load attempts will fail (because containing folder does not exists) however we still need to record all failed lookup locations.
      */
     function loadModuleFromFile(extensions: Extensions, candidate: string, failedLookupLocations: Push<string>, onlyRecordFailures: boolean, state: ModuleResolutionState): PathAndExtension | undefined {
+        if (extensions === Extensions.Json) {

```

Comment:
Seems like this could be handled along with the bottom case `if (hasJavaScriptFileExtension)`?